src/Entity/OgMembership.php

```diff
@@ -181,31 +187,42 @@ public function getType() {
   /**
    * {@inheritdoc}
    */
-  public function setRoles($role_ids) {
-    $this->set('roles', $role_ids);
+  public function setRoles(array $roles = array()) {
+    $role_ids = array_map(function (OgRole $role) {
+      return $role->id();
+    }, $roles);
+
+    $this->set('roles', array_unique($role_ids));
```

Comment:
Small suggestion: maybe move this method underneath `::getRoles()` so that it is nicely grouped together, like you did with the other methods that you worked on in this PR?